ACadSharp/Color.cs

```diff
@@ -5,297 +5,445 @@
-			{176, new byte[] {0, 0, 76}},
-			{177, new byte[] {38, 38, 126}},
-			{178, new byte[] {0, 0, 38}},
-			{179, new byte[] {19, 19, 88}},
-			{180, new byte[] {63, 0, 255}},
-			{181, new byte[] {159, 127, 255}},
-			{182, new byte[] {41, 0, 165}},
-			{183, new byte[] {103, 82, 165}},
-			{184, new byte[] {31, 0, 127}},
-			{185, new byte[] {79, 63, 127}},
-			{186, new byte[] {19, 0, 76}},
-			{187, new byte[] {47, 38, 126}},
-			{188, new byte[] {9, 0, 38}},
-			{189, new byte[] {23, 19, 88}},
-			{190, new byte[] {127, 0, 255}},
-			{191, new byte[] {191, 127, 255}},
-			{192, new byte[] {165, 0, 82}},
-			{193, new byte[] {124, 82, 165}},
-			{194, new byte[] {63, 0, 127}},
-			{195, new byte[] {95, 63, 127}},
-			{196, new byte[] {38, 0, 76}},
-			{197, new byte[] {57, 38, 126}},
-			{198, new byte[] {19, 0, 38}},
-			{199, new byte[] {28, 19, 88}},
-			{200, new byte[] {191, 0, 255}},
-			{201, new byte[] {223, 127, 255}},
-			{202, new byte[] {124, 0, 165}},
-			{203, new byte[] {142, 82, 165}},
-			{204, new byte[] {95, 0, 127}},
-			{205, new byte[] {111, 63, 127}},
-			{206, new byte[] {57, 0, 76}},
-			{207, new byte[] {66, 38, 76}},
-			{208, new byte[] {28, 0, 38}},
-			{209, new byte[] {88, 19, 88}},
-			{210, new byte[] {255, 0, 255}},
-			{211, new byte[] {255, 127, 255}},
-			{212, new byte[] {165, 0, 165}},
-			{213, new byte[] {165, 82, 165}},
-			{214, new byte[] {127, 0, 127}},
-			{215, new byte[] {127, 63, 127}},
-			{216, new byte[] {76, 0, 76}},
-			{217, new byte[] {76, 38, 76}},
-			{218, new byte[] {38, 0, 38}},
-			{219, new byte[] {88, 19, 88}},
-			{220, new byte[] {255, 0, 191}},
-			{221, new byte[] {255, 127, 223}},
-			{222, new byte[] {165, 0, 124}},
-			{223, new byte[] {165, 82, 145}},
-			{224, new byte[] {127, 0, 95}},
-			{225, new byte[] {127, 63, 111}},
-			{226, new byte[] {76, 0, 57}},
-			{227, new byte[] {76, 38, 66}},
-			{228, new byte[] {38, 0, 28}},
-			{229, new byte[] {88, 19, 88}},
-			{230, new byte[] {255, 0, 127}},
-			{231, new byte[] {255, 127, 191}},
-			{232, new byte[] {165, 0, 82}},
-			{233, new byte[] {165, 82, 124}},
-			{234, new byte[] {127, 0, 63}},
-			{235, new byte[] {127, 63, 95}},
-			{236, new byte[] {76, 0, 38}},
-			{237, new byte[] {76, 38, 57}},
-			{238, new byte[] {38, 0, 19}},
-			{239, new byte[] {88, 19, 28}},
-			{240, new byte[] {255, 0, 63}},
-			{241, new byte[] {255, 127, 159}},
-			{242, new byte[] {165, 0, 41}},
-			{243, new byte[] {165, 82, 103}},
-			{244, new byte[] {127, 0, 31}},
-			{245, new byte[] {127, 63, 79}},
-			{246, new byte[] {76, 0, 19}},
-			{247, new byte[] {76, 38, 47}},
-			{248, new byte[] {38, 0, 9}},
-			{249, new byte[] {88, 19, 23}},
-			{250, new byte[] {0, 0, 0}},
-			{251, new byte[] {101, 101, 101}},
-			{252, new byte[] {102, 102, 102}},
-			{253, new byte[] {153, 153, 153}},
-			{254, new byte[] {204, 204, 204}},
-			{255, new byte[] {255, 255, 255}}
-		};
+    public readonly struct Color
```

Comment:
A are we sure that we want to use a `readonly struct`?? This may simplify the implementation for the development side but lacks flexibility to mutate the struct.

I'm not against I just want to know the reason for this decision.

Reply:
I have to dig again to find some of the articles, but essentially having mutable structs can have unintended consequences during usages.  My thoughts are if this is headed off at the pass with a `readonly struct` then headaches can be avoided later. See the two articles below.
https://stackoverflow.com/a/441323
https://learn.microsoft.com/en-us/archive/blogs/ericlippert/mutating-readonly-structs

I'm honestly open to changing it back to the way you originally had it since it is effectively a struct acting as a simple wrapper for an int32. I was just trying to avoid potential foot guns for later.

I'll await your decision.

Reply:
Keep it this way, as I said, I know that this makes implementing way easier, but we have to keep in mind that Autocad Color also has the ColorName field, check the DxfCode 430, but is not tested nor implemented, but I'm guessing that is for colors that are not in the dictionary so the RGB values may need to be open for change in future implementations.

Reply:
This is a good point.  I didn't even know was a thing in the design.  I've been working with DWGs and DXFs for a while but I've not had a need of a named color before other than the pseudo named red, green, blue colors that ACAD translates to indexed colors.  I'll try to run through my bank of DWGs and see if any of them use this code.